freud/box.pyx

```diff
@@ -301,9 +302,13 @@ cdef class Box:
         vecs = np.atleast_2d(vecs)
         vecs = freud.util._convert_array(vecs, shape=(None, 3)).copy()
```

Comment:
Notice that we're copying the input vectors here. That means that any change you make is still going to be to a copy of the input arrays. You need to change whether or not a copy is made based on the `inplace` argument.

Furthermore, the `_convert_array` function (which you can find in `freud/util.pyx`) may internally copy the array if it decides it has to (look at the call to `numpy.require` in that function). That means you also need to modify `_convert_array` to basically throw an error in the `inplace` case if the requirements aren't satisfied. I'm happy to give you more information on that offline if you'd like, or you can try to read the docs for these functions and try to understand them a bit more and come back with specific questions, up to you!

Reply:
Yeah that makes sense. I didn't read that carefully and thought vecs = freud.util._convert_array(). Let me try to fix that